lib/mongodb.ts

```diff
@@ -7,8 +7,8 @@ if (!process.env.MONGODB_URI) {
 const uri = process.env.MONGODB_URI;
 const options = {};
 
-let client;
-let getDBClient;
+let client: MongoClient;
+let getDBClient: Promise<MongoClient>;
```

Comment:
ok. a bit tricky comment. 

While this is working fine, we need to improve it. 
Because returning "Promise" is not a good pattern here because if you return Promise, the place where you use it is like the following
```
         const client = await getDBClient;
```
in the usage above, `getDBClient` got already a promise without calling part, `()`.


What it should look like is `const client = await getDBClient()`. Because the naming `getDBClient` suggests it's a function. In order to do so, we need to type `getDBClient` `() => Promise<MongoClient>` rather than `Promise<MongoClient>`.

Reply:
![Screenshot 2023-03-14 at 15 57 23](https://user-images.githubusercontent.com/60405643/225042107-395e88d5-5df0-4a1a-8e85-c5091c78770f.png)

But then I get this error message...
![Screenshot 2023-03-14 at 15 57 34](https://user-images.githubusercontent.com/60405643/225042134-de8b9dfc-7869-4814-b8a3-44506f9a9647.png)

Reply:
so in that case, we can just pass the function itself, (without calling)
```
getDBClient = client.connect
```

Reply:
hmm I can not get/post any data now, after I changed type to `() => Promise<MongoClient>` and calling getDBClient with `const client = await getDBClient()` when to use.

Reply:
ok. I think there is something wrong with the timing when should create a client. 
```
const getDBClient = async (): Promise<MongoClient> => {
  const client = new MongoClient(uri, options);
  client.connect();
  return client;
};
```
try this. I found the official example: https://www.mongodb.com/compatibility/using-typescript-with-mongodb-tutorial